gax/src/main/java/com/google/api/gax/retrying/CallbackChainRetryingFuture.java

```diff
@@ -52,14 +52,15 @@
  * <p>This class is thread-safe.
  */
 class CallbackChainRetryingFuture<ResponseT> extends BasicRetryingFuture<ResponseT> {
-  private final RetryingExecutor<ResponseT> retryingExecutor;
+  private final ScheduledRetryingExecutor<ResponseT> retryingExecutor;
```

Comment:
Why was this changed to `ScheduledRetryingExecutor`?

Reply:
Because this class is very tightly coupled to that class’ behavior. pretending that any other implementation can fit doesn’t make sense. Also I’m trying to minimize the references to retryingexecutorwithcontext, because it’s slated to be removed with a java8 upgrade